### DISCUSSION TITLE
Posting raw json

### DISCUSSION THREAD

**Discussion by bananashell** [datalust/seq-tickets#1206] [Help] [answered]
Hi,

I have a dotnet core middleware that handles all logging of incoming requests.
From there I want to log body as Json to SEQ.

The jist of it 

```C#
var enrich = new Dictionary<string, object>
    {
        { "@Body", "{\"hello\": \"world\"}"},
    };

logger.BeginScope(enrich){
    await next(context);
}

```

Is there a way to tell SEQ that value is already in JSON and no serialization is needed?

**Comment by nblumhardt** [comment, marked as answer]
Hi Joakim, thanks for getting in touch.

Unfortunately, no - not without deserializing the JSON, and then logging the deserialized object. Historically we've avoided adding this because of the difficulty of preventing invalid JSON from making its way into the pipeline.

On the Seq side, though, if you need to manipulate JSON you can use `FromJson()`, e.g. `FromJson(Body).hello = 'world'` etc.

Hope this helps,
Nick

**Reply by bananashell** [reply]
Hmm ok, too bad.
Are there issues with validating with System.Text.Json?

Kudos on the quick response 👍🏻

**Reply by nblumhardt** [reply]
Thanks for the reply. Only that the cost would (currently, I think?) be comparable to just deserializing the JSON before logging the deserialized object. Could be one for us to think about in the future, though.
